### ISSUE TITLE
saving additional metadata with each revision

### ISSUE THREAD

**Issue by dkharrat** [bdurand/acts_as_revisionable#4]
Is there a simple way to save metadata with each revision (e.g. user who initiated the change, or the ip address of the user, etc.)? It would be nice if the gem can support such custom data, similar to how the [paper_trail](https://github.com/airblade/paper_trail) gem supports it.

**Comment by bdurand** [comment]
The latest version of the gem (1.1.2):

Revision records have an optional label field which can be used for display purposes to distinguish revisions n a view. This value will only be set if you provide a Proc for the :label option to the acts_as_revisionable call. The proc will be yielded to with the record before it is revisioned.

**Comment by dkharrat** [comment]
Yeah, I did notice that, and it's a nice feature. It works well when you want to show some text to the user for the revision. However, that still doesn't really work in my case. I was looking for a way to save custom data (e.g. user_id, ip address, etc.) via additional columns in the revisions table, in order to facilitate things like searching and filtering by SQL queries. For example, "finding all the revisions created by a specific user".

**Comment by bdurand** [comment]
What do you think about just extending the feature so that you can define your own fields on RevisionRecord and provide the values with a proc option on acts_as_revisionable similar to how label gets set?

**Comment by dkharrat** [comment]
That sounds good. I recommend removing the label column and instead let the user explicitly define it using this mechanism. That way, those who don't need a 'label' column don't need to be forced to have an additional unused column in the database.

I also suggest the following syntax (with the requirement being that the columns must be manually added to the revisions table):

```
acts_as_revisionable :meta => { :label      => lambda { |record| "#{record.name} updated" },
                                :word_count => :count_words,
                                :answer     => 42 }
```

In the above example, `count_words` is a method defined in the model.

**Comment by bdurand** [comment]
That's just what I was thinking.

I just released version 1.2.0 with this feature in it as well as support for subclassing RevisionRecord in case you want to add your own logic or separate some revisions into another table.

**Comment by dkharrat** [comment]
This is awesome! Thanks for the quick turnaround in implementing this cool feature!
